tests/test_variant.cc

```diff
@@ -165,3 +165,23 @@ TEST_CASE("variant: visit()", "[variant]")
     v = 42L;
     REQUIRE(gul14::visit(get_type_name, v) == "long");
 }
+
+TEST_CASE("make_overload_set()", "[variant]")
+{
+    auto v = gul14::variant<float, double, long>{ };
+
+    auto get_type_name = gul14::make_overload_set(
+        [](auto&&) -> std::string { return "FAIL"; },
```

Comment:
Can we have also a test case for the failure case? E.g:
```c++
auto w = gul14::variant<bool, std::string>{ };
REQUIRE(gul14::visit(get_type_name, w) == "FAIL");
w = false;
REQUIRE(gul14::visit(get_type_name, w) == "FAIL");
```

Reply:
Good idea. I have added a test for the `auto&&` case.